### DISCUSSION TITLE
Solcx Installation Problems

### DISCUSSION THREAD

**Discussion by JohnEMagee** [smartcontractkit/full-blockchain-solidity-course-py#1192] [Q&A] [answered]
I have read the previous discussion and have found nothing useful for this issue on the SolcX and have not found any solutions that actually help.

I'm running a Mac M1 Mini, Python 3.9.10 (64 bit) and **can not** install solc anyway.  Looking at the build it is listed currently as failing as well...I have checked that it installed properly in _/opt/homebrew/lib/python3.9/site-packages_.  There's only one login on my computer as it's mine so it's also the admin (and this should be able to work on the admin)

If the only solution is a virtual environment it would be helpful if someone could point me to a good tutorial because i was reading that you can't use venv to use different python versions any more in python 3.

**Comment by cromewar** [comment]
Hello @JohnEMagee 
You can install any python version you want and in order to select that version for a virtualenv (not venv which is a different package) you can use something like this:

`virtualenv --python=/usr/bin/python3.x NameOfTheEnv`

Also, as you are using a M1 Mac those have problems with web3 libraries yet, does [this](https://ethereum.stackexchange.com/questions/84125/brownie-cannot-install-solc-on-osx) help you?

**Reply by JohnEMagee** [reply]
This is almost perfect BUT you have to do a special trick to install solidity 6 and while the stack exchange has an answer you can try and follow there's a better answer for installing previous versions of solidity [at the repository itself](https://github.com/ethereum/homebrew-ethereum#solidity)

However this doesn't solve the problem, it does not run the binary (0.6.12) so suggest compile solc instead of install solc - and this fails as well...so I tried the 3rd suggestion of using the 3rd party binary, and renaming it and moving it, and then THAT didn't work cause I couldn't execute it, so then I changed all the permissions to execute and i still got an error of Bad CPU Type.

At this point, I've followed all the steps provided by the github and a few other suggestions and none of them worked.  It's pretty disappointing I've gotten to this step, maybe I'll try the virtualenv but now I have to spend another hour I could have been studying fixing this...whereas this has probably been a problem for a while so it would have been nice to have some forewaring i was going to get 3.5 hours into a tutorial and then run into a massive road block

**Comment by JohnEMagee** [comment, marked as answer]
The only easy solution I've found so far is that I can compile solidity 0.8.12, it fully compiled.  I used solc version 0.8.12 and changed the pragma in the SimpleStorage file to 0.8.12

I did try various ways to compile at least 0.6.12 because you can still use homebrew to download it, but none of them worked.  

You have to COMPILE 0.8.12 (and change all your references) the first time (but you can comment it out afterwards thankfully)...but I haven't moved forward beyond this issue today because I spent most of my free study time figuring this out...it seems the m1 isn't available yet for testing and such based on some things I've read, hopefully this changes...soon

**Reply by cromewar** [reply]
Nice catch
